### ISSUE TITLE
Make it possible to write scheme libraries using markdown

### ISSUE THREAD

**Issue by einkoder** [letloop/cli#1747]
Execute the code blocks of a markdown file, one after the other.

**Comment by amirouche** [comment]
Thanks for your interest. 

I am not sure it mesh very well with the current offering. If it is possible to pack libraries inside the binary, to create a complete scheme distribution, it can make sense.

Do you have more precise specifications in mind?

**Comment by amirouche** [comment]
# mockup

Given the following markdown file:

---

# Library `(letloop base)`

```scheme
(import (only (chezscheme) define lambda))
```

## Procedure `(ruse)` → `procedure?` 

```scheme
(define ruse
  (lambda () ruse))
```

```scheme
(define ~check-ruse-000-ruse-is-ruse
  (lambda () (eq? ruse (ruse))))
```

---

The markdown file will be evaluated to:

```scheme
(library (letloop base)
  (export ruse)
  (import (only (chezscheme) lambda define))

  (define ruse (lambda () ruse))

  (define ~check-ruse-000-ruse-is-ruse
    (lambda () (eq? ruse (ruse))))
```

**Comment by amirouche** [comment]
@einkoder WDYT?

**Comment by einkoder** [comment]
That is not a program notebook, but a library written in a markdown file. Support for macro would be nice.

**Comment by amirouche** [comment]
> That is not a program notebook, but a library written in a markdown file.

Indeed. Given `letloop check` reduces the need for a good REPL, in my experience, I save time when I work inside a library.

The following will be supported:

```scheme
#;letloop> cat my-library-notebook.md
# Library `(my-library-notebook)`

## Procedure `(main)`

...

#;letloop> cat my-program-notebook.scm
(import (chezscheme))
(import (my-library-notebook))

(main)
#;letloop> letloop exec . my-program-notebook.scm
```

At some point, I will make it easier to run thunks directly from the command line, so the above will before:

```
letloop my-library-notebook.md main
```

> Support for macro would be nice.

Indeed.

**Comment by amirouche** [comment]
I merged https://github.com/letloop/cli/pull/1821 with initial markdown support.

**Comment by amirouche** [comment]
Dear @einkoder, 

Literally is a significant feature 🥲.

I have to pull out the feature from the next release, because of the dependency on cmark that gives work to compile on all supported OS. I will reconsider its inclusion later once if: 

- There is a good enough CommonMark parser (https://commonmark.org/) that does not rely on code that requires something else but Chez to compile;
- There is a working (Just Linux) aka. NixOS-like distro with Alpine and Chez.

Keep it steady!

NB: Nice hair cut.
